docs/migration.md

```diff
@@ -0,0 +1,453 @@
+        <groupId>org.finos.symphony.bdk</groupId>
+        <artifactId>symphony-bdk-http-jersey2</artifactId> <!-- or symphony-bdk-http-webclient -->
+        <scope>runtime</scope>
+    </dependency>
+    <dependency>
+        <groupId>org.finos.symphony.bdk</groupId>
+        <artifactId>symphony-bdk-template-freemarker</artifactId>  <!-- or symphony-bdk-http-handlebars -->
+        <scope>runtime</scope>
+    </dependency>
+</dependencies>
+```
+</td>
+</tr>
+</table>
+
+## Bot's configuration
+In Java BDK 1.0, two configuration files were required : `application.yaml` (or `application.config`) and `bot-config.json`. Java BDK 2.0 lightened the configuration. Therefore, only `src/main/resources/config.yaml` file is required with a minimum of configuration.
+
+Bot’s configuration in Java BDK 2.0 should have the following properties:
+- `host`: pod’s host name
+
+- `bot.username`: bot’s (or service account) username
+
+- `bot.privatekey.path`: bot’s private key path
+
+If your bot is deployed on premise, the following properties are required as well:
+
+- `agent`: on premise agent configuration
+
+- `keyManager`: on premise Key manager configuration
+
+- `proxy`: proxy configuration to reach the pod
+
+- `ssl.trustStore`: trust store path and password
+
+> Click [here](./configuration.md) for more detailed documentation about BDK configuration
+
+### Minimal configuration example
+#### Spring Boot based project
+<table>
+<tr>
+<th>Java BDK 1.0</th>
+<th>Java BDK 2.0</th>
+</tr>
+<tr>
+<td>
+
+#### **`application.yaml`:**
+```yaml
+server:
+    port: 8080
+    servlet:
+        context-path: "/botapp"
+
+certs: /path/to/private/key
+bot-config: /path/to/bot-config.json
+```
+
+#### **`bot-config.json`:**
+```json
+{
+  "sessionAuthHost": "session.symphony.com",
+  "sessionAuthPort": 443,
+  "keyAuthHost": "km.symphony.com",
+  "keyAuthPort": 443,
+  "podHost": "pod.symphony.com",
+  "podPort": 443,
+  "agentHost": "agent.symphony.com",
+  "agentPort": 443,
+  "botPrivateKeyPath": "certs/",
+  "botPrivateKeyName": "/path/to/bot/rsa-privatekey.pem",
+  "botUsername": "bot-username",
+  "authTokenRefreshPeriod": "30",
+  "authenticationFilterUrlPattern": "/secure/",
+  "showFirehoseErrors": false,
+  "connectionTimeout": 45000, 
+  "proxyURL": "proxy.symphony.com",
+  "proxyUsername": "proxy.username",
+  "proxyPassword": "proxy.password",
+  "keyManagerProxyURL": "km.proxy.symphony.com",
+  "keyManagerProxyUsername": "km.proxy.username",
+  "keyManagerProxyPassword": "km.proxy.password"
+}
+```
+</td>
+<td>
+
+Only `config.yaml`  file is required. It can be in *JSON* 
+
+
+```json
+{
+    "host": "acme.symphony.com",
+    "bot": {
+        "username": "bot-username",
+        "privateKey": {
+            "path": "/path/to/bot/rsa-privatekey.pem"
+        }
+    },
+    "proxy": {
+        "host": "proxy.symphony.com",
+        "port": 1234,
+        "username": "proxyuser",
+        "password": "proxypassword"
+    },
+    "keyManager": {
```

Comment:
could you add the `proxy` field under `keyManager` as put in the yaml example below?